File: C#/16-Ordenação de Vetores/SelectionSort.cs
using System;

class SelectionSort
{
    //Método selection sort para ordenação de vetores
    public static int[] selectionSort(int[] vetor) 
    {
        int min;
        int aux;

        for (int i = 0; i < vetor.Length - 1; i++) 
        {
            min = i;

            for (int j = i; j < vetor.Length; j++) 
            {
                if (vetor[j] < vetor[min]) 
                {
                    min = j;
                }

                if (min != i)
                {
                    aux = vetor[min];
                    vetor[min] = vetor[i];
                    vetor[i] = aux;
                }
            }
        }

        return vetor;
    }
    static void Main(string[] args)
    {
        int[] numeros = new int[] { 2, 5, 1, 3, 4 };
        Console.WriteLine("Vetor original:");
        foreach (int numero in numeros)
        {
            Console.WriteLine(numero);
        }

        Console.WriteLine("Pós selection sort");

        //Vamos invocar o método passando vetor numeros como argumento
        selectionSort(numeros);
        //Exibindo vetor ordenado
        foreach (int numero in numeros)
        {
            Console.WriteLine(numero);
        }

        Console.ReadLine();
    }
}

File: Java/14-Ordenação de Vetores/SelectionSort.java
class SelectionSort {
   // Podemos ordenar um vetor através de alguns métodos criados por programadores
    // Este método é chamado SelectionSort
    static int[] selectionSort(int[] vetor){
       int min;
       int aux;
       
       for(int i = 0; i < vetor.length; i++) {
           min = i;
           
           for(int j = i; j < vetor.length; j++) {
               
               if(vetor[j] < vetor[min]) {
                   
                   min = j;
               }
               
               if(min != i) {
                   aux = vetor[min];
                   vetor[min] = vetor[i];
                   vetor[i] = aux;
               }
           }
       }
       
       return vetor;
    }
    public static void main(String[] args) {
        // Vetor esta fora de ordem
        int[] vet = { 4, 2, 1, 8, 3 };

        System.out.println("Vetor original");
        for (int v : vet) {
            System.out.print(v + " ");
        }

        System.out.println();

         // Vamos invocar o método passando vetor vet como argumento
         selectionSort(vet);
         System.out.println("Apos SelectionSort");
         for (int v : vet) {
             System.out.print(v + " ");
         }
    }
}